### ISSUE TITLE
"license: unknown-license" badge on repository page

### ISSUE THREAD

**Issue by vi** [brson/error-chain#67]
Yet `Cargo.toml` contains licence field:

> license = "MIT/Apache-2.0"

**Comment by Yamakaky** [comment]
Yeah, I don't know why... Maybe then only check the `LICENSE` file.

**Comment by colin-kiegel** [comment]
That seems to be correct.
a) shields seems to use the github API (https://github.com/badges/shields/blob/b74107eb3d4301c192e5dfd6d252e8a3b90d16bf/suggest.js#L36)
b) the github API seems to rely on a license file (https://developer.github.com/v3/licenses/#get-a-repositorys-license)

**Comment by vi** [comment]
Also Github seems to show licence in tabs (Example "Apache 2.0" in https://github.com/rust-lang/cargo/ )

>  3,571 commits 5 branches    17 releases    264 contributors    Apache-2.0

making such shields redundant.

**Comment by joshtriplett** [comment]
Ideally, someone should update github's license detection to look at Cargo metadata too.

**Comment by vi** [comment]
You can report it to [Github feedback page](https://github.com/contact/). They have internal issue/feature request tracker and always reply something.

**Comment by nelsonjchen** [comment]
GitHub documentation [says](https://help.github.com/articles/adding-a-license-to-a-repository/#detecting-a-license) they're using the [licensee ruby gem](https://github.com/benbalter/licensee). While they do have some heuristics for detecting fields of package managers (e.g. [gemspec](https://github.com/benbalter/licensee/blob/38e3cdeffae0dd226945bbeda4d77a7b9fa4cb49/lib/licensee/matchers/gemspec_matcher.rb), [bower](https://github.com/benbalter/licensee/blob/38e3cdeffae0dd226945bbeda4d77a7b9fa4cb49/lib/licensee/matchers/gemspec_matcher.rb)), licensee appears to [recommend just sticking the license text in the repository and not relying upon the field of the package manager](https://github.com/benbalter/licensee/blob/master/docs/what-we-look-at.md#why-not-just-look-at-the-license-field-of-insert-package-manager-here). I'm guessing those language ecosystem specific heuristics are a bit old and from a long time ago and are grandfathered in. [I don't see CPAN,  D, Elm, or really any wide assortment of matchers for ecosystems in their matchers][1]. 

Licensee could be updated, but I would wager that the request won't be accepted. 

[1]:https://github.com/benbalter/licensee/tree/38e3cdeffae0dd226945bbeda4d77a7b9fa4cb49/lib/licensee/matchers

**Comment by nelsonjchen** [comment]
Or... maybe they do. I guess I just wasn't looking hard enough. That said, they do say the R people explicitly recommend not including the license text.

https://github.com/benbalter/licensee/pull/129
